docs/low-code-spark/extensibility/gem-builder.md

```diff
@@ -6,20 +6,857 @@ description: Gem-Builder
 tags: []
 ---
 
-Prophecy Data Engineering system can be extended with **Gem Builder** to add **your own standard sources, targets and
-transforms** to Spark and roll it out to your entire team. You can build **your own frameworks with** custom Data Quality
-library, or auditing library.
+:::caution Enterprise Only
+
+Please [contact us](https://www.prophecy.io/request-a-demo) to learn more about the Enterprise offering.
+
+:::
+
+[Gems](docs/concepts/gems.md) are operations you can add to your Pipeline to perform actions on your data. The **Gem Builder** allows you to create custom UI experiences that will generate optimized code wherever it's used. You can add your own custom sources, targets, and transformations and roll them out to your entire team. You can even build a custom Data Quality library, or an auditing library.
 
 <div class="video-container">
 <iframe src="https://www.youtube.com/embed/K23pOatAeVE" title="YouTube video player" frameborder="0"
 allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen>
 </iframe>
 </div>
-
 <br />
 
-:::caution Learn more
+## Getting Started
+
+All of our Gems are stored at the account level. Once you navigate to the Gem listing page you will see all the Prophecy-defined and User Defined Gems. You can add a new Gem or modify an existing one.
+
+After writing your Gem code, you can click `Preview` to see how the Gem UI looks. You can fill in some values, and click save to check the Spark code which will be generated for the Gem.
+Finally, just publish the Gems and your Gem is ready to be used in the Pipelines.
+
+Please refer below video for a step-by-step example:
+
+<div class="wistia_responsive_padding" style={{padding:'56.25% 0 0 0', position:'relative'}}>
+<div class="wistia_responsive_wrapper" style={{height:'100%',left:0,position:'absolute',top:0,width:'100%'}}>
+<iframe src="https://user-images.githubusercontent.com/121796483/212867119-a05b1204-0be7-4f0b-866b-675b4b0b4280.mp4" title="Gem builder" allow="autoplay;fullscreen" allowtransparency="true" frameborder="0" scrolling="no" class="wistia_embed" name="wistia_embed" msallowfullscreen width="100%" height="100%"></iframe>
+</div></div>
+
+## Tutorial
+
+The Gem builder is a tool that enables users to create any custom Gems or modify existing ones. There are two types of Gems:
+
+- **DataSource Gems**: These Gems enable the reading and writing of data from or to various data sources
+- **Transform Gems**: These Gems apply transformations/joins/any other custom logic onto any DataFrame(s) that are passed into them.
```

Comment:
The above text doesn't match the UI: 
<img width="474" alt="image" src="https://user-images.githubusercontent.com/20563764/213931839-d5869211-d5ad-4824-9416-6416e8672279.png">


According to this UI, it would appear that I can create a Gem.Gem or a Gem.DataSource. 
Furthermore, either Gem.Gem or Gem.DataSource could be a Custom, Transform, Join/Split, MyGem, or Hidden. 

The text is much clearer than the UI...can the UI be clarified?
Perhaps the user needs to select 
Create custom Gem!
Choose: should this gem be shared? If so, select teams (or projects?) that should be able to use this Custom Gem.
Choose: DataSource, DataTarget, or Transformation
If Transformation, 
Choose: Join/Split or Other

Reply:
Edit: below the DataSource is called a `Dataset` ... so we need to make this consistent.

Reply:
Updating the screenshot with new UI.